dcos/http.py

```diff
@@ -180,3 +220,29 @@ def delete(url, **kwargs):
 def silence_requests_warnings():
     """Silence warnings from requests.packages.urllib3.  See DCOS-1007."""
     requests.packages.urllib3.disable_warnings()
+
+
+def _get_http_auth_credentials(response):
```

Comment:
the effect of the function is that it sets the value in the dictionary.

If you'd like to keep it `get`, return the credentials, and set them in the caller, then that would be fine (and probably better to keep things as functional as possible).